pkg/sdkserver/sdkserver.go

```diff
@@ -762,20 +790,202 @@ func (s *SDKServer) GetCounter(ctx context.Context, in *alpha.GetCounterRequest)
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement me
-	return nil, errors.Errorf("Unimplemented -- GetCounter coming soon")
+
+	s.logger.WithField("name", in.Name).Info("Getting Counter")
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	counter, ok := gs.Status.Counters[in.Name]
+	if !ok {
+		return nil, errors.Errorf("NOT_FOUND. %s Counter not found", in.Name)
```

Comment:
This error formatting (and the rest in this file) isn't the Google style guide for go. See:
https://google.github.io/styleguide/go/decisions.html#error-strings

```suggestion
		return nil, errors.Errorf("count not found: %s", in.Name)
```

The errors package includes stacktracking, so it does report where errors come from.

Reply:
This one's following the https://developers.google.com/actions-center/reference/grpc-api/status_codes :) I'll change it to match the rest of the file.

Reply:
I am wondering of this might give you what you want: https://pkg.go.dev/google.golang.org/grpc/status

Reply:
Yes? Although I vaguely remember a conversation about not using a different error package (can't find where we had that though). It would be `status.Errorf(codes.NotFound, "%s Counter not found", in.Name)` instead of the `errors.Errorf`. Do we have any preference on keeping the current errors package and format, or moving the CountsAndLists to the grpc status errors?

Reply:
You ask a very good question - we've not done gRPC status values anywhere else in the codebase.

Although maybe should for gRPC endpoints? And this could be the beginning of the efforts. I feel I'm the worst at sticking to API standards 😄 , so WDYT? @zmerlynn @gongmax thoughts?

Reply:
@igooch I think you and I previously had a review back-and-forth over just staying consistent. I think consistency of error return types from the API wins over API guidelines. (But I haven't revewed this exact one yet, just stating my general feeling.)

Reply:
So are we leaning towards Mark's original suggestion of `return nil, errors.Errorf("count not found: %s", in.Name)`?

Reply:
Honestly, I don't have strong opinions either way (original with lower case or use gRPC errors).

I've not seen any major downside to using `return nil, errors.Errorf("count not found: %s", in.Name)?` with gRPC though - but very happy to hear dissenting opinions.